File: Resolução em python/ex6.py
total_alunos = 5
qtdfinal = 0

for aluno in range(1, total_alunos + 1):
  nota1 = float(input("Informe a primeira nota do aluno {}: ".format(aluno)))
  while nota1 < 0 or nota1 > 10:
    print("Nota inválida. Informe uma nota entre 0 e 10.")
    nota1 = float(input("Informe a primeira nota do aluno {}: ".format(aluno)))

  nota2 = float(input("Informe a segunda nota do aluno {}: ".format(aluno)))
  while nota2 < 0 or nota2 > 10:
    print("Nota inválida. Informe uma nota entre 0 e 10.")
    nota2 = float(input("Informe a segunda nota do aluno {}: ".format(aluno)))

  media = (nota1 + nota2) / 2

  if media >= 2 and media < 6:
    qtdfinal += 1

percent = (qtdfinal / total_alunos) * 100

print("Percentual de alunos em Prova Final: {:.2f}%".format(percent))

File: Resolução em C/ex6.c
#include <stdio.h>

int ex6() {
  int numAlunos = 5;
  float notas[numAlunos];
  char nomes[numAlunos][50];
  int contFinal = 0;

  for (int i = 0; i < numAlunos; i++) {
    // Entrada de dados
    printf("Informe o nome do aluno %d: ", i + 1);
    scanf("%s", nomes[i]);

    do {
      printf("Informe a primeira nota do aluno %s: ", nomes[i]);
      scanf("%f", &notas[i]);
      if (notas[i] < 0 || notas[i] > 10) {
        printf("Nota inválida. Digite uma nota entre 0 e 10.\n");
      }
    } while (notas[i] < 0 || notas[i] > 10);

    do {
      printf("Informe a segunda nota do aluno %s: ", nomes[i]);
      scanf("%f", &notas[i]);
      if (notas[i] < 0 || notas[i] > 10) {
        printf("Nota inválida. Digite uma nota entre 0 e 10.\n");
      }
    } while (notas[i] < 0 || notas[i] > 10);

    float media = (notas[i] + notas[i]) / 2;

    if (media < 2) {
      printf("Aluno %s: Reprovado\n", nomes[i]);
    } else if (media < 6) {
      contFinal++;
      printf("Aluno %s: Prova Final\n", nomes[i]);
    } else {
      printf("Aluno %s: Aprovado\n", nomes[i]);
    }
  }

  float percen = (float)contFinal / numAlunos * 100;

  printf("Percentual de alunos em Prova Final: %.2f%%\n", percen);

  return 0;
}